pydantic/_internal/_fields.py

```diff
@@ -22,6 +22,19 @@ def get_type_hints_infer_globalns(
     localns: dict[str, Any] | None = None,
     include_extras: bool = False,
 ) -> dict[str, Any]:
+    """Gets type hints for an object by inferring the global namespace.
+
+    It uses the `typing.get_type_hints`, The only thing that we do here is fetching
+    global namespac from `obj.__module__` if it is not `None`.
```

Comment:
@hramezani A bit late: `namespace`

Reply:
Thanks
PR is welcome if you would like to fix it. Otherwise If will create a PR

Reply:
Will do